src/ResourceDatagramSocket.php

```diff
@@ -68,7 +68,7 @@ public function __construct($socket, int $limit = self::DEFAULT_LIMIT)
             } else {
                 EventLoop::disable($callbackId);
 
-                $reader->resume([ResourceSocketAddress::fromString($address), $data]);
+                $reader->resume([SocketAddressType::fromString($address), $data]);
```

Comment:
I read that as returning a `SocketAddressType` instead of a `SocketAddress`, so should we really put the methods on the enum?

Reply:
How about functions in the `Amp\Socket\SocketAddress` namespace? That would become `SocketAddress\fromString()`.

Reply:
Functions might be a good idea, yeah.